src/Tribe/Template/Month.php

```diff
@@ -288,6 +279,40 @@ private static function view_more_link( $date ) {
 			return esc_url_raw( $day_link );
 		}
 
+
+		/**
+		 * Get all the events in the month by directly querying the postmeta table
+		 * Also caches the postmeta and terms for the found events
+		 */
+		protected function set_events_in_month() {
+			global $wpdb;
+
+			// note: this query returns the event dates as timestamps, makes date comparison faster later
+			$events_request =
+				$wpdb->prepare(
+					"SELECT tribe_event_start.post_id as ID,
+							UNIX_TIMESTAMP(tribe_event_start.meta_value) as EventStartDate,
+							UNIX_TIMESTAMP( tribe_event_end_date.meta_value) as EventEndDate
+					FROM $wpdb->postmeta AS tribe_event_start
+					LEFT JOIN $wpdb->postmeta as tribe_event_end_date ON ( tribe_event_start.post_id = tribe_event_end_date.post_id AND tribe_event_end_date.meta_key = '_EventEndDate' )
+					WHERE tribe_event_start.meta_key = '_EventStartDate'
+					AND ( (tribe_event_start.meta_value >= '%1\$s' AND  tribe_event_start.meta_value <= '%2\$s')
+						OR (tribe_event_start.meta_value <= '%1\$s' AND tribe_event_end_date.meta_value >= '%1\$s')
+						OR ( tribe_event_start.meta_value >= '%1\$s' AND  tribe_event_start.meta_value <= '%2\$s')
+					)
+					",
+					tribe_event_beginning_of_day($this->first_grid_date),
+					tribe_event_end_of_day($this->final_grid_date)
+			);
+			$this->events_in_month = $wpdb->get_results( $events_request );
```

Comment:
Since we are opting away from using `WP_Query` (a reasonable decision), it feels like the result of this query should be cached. Maybe something like:

```php
wp_cache_set( $month, $this->events_in_month, 'tribe-events-in-month', MINUTE_IN_SECONDS * 30 );
```

Then we can manually purge the cache for the month (or for two months if the event's date is moved drastically) when an event is saved.